core/adapters/eth_tx.go

```diff
@@ -42,6 +44,93 @@ func (e *EthTx) TaskType() models.TaskType {
 // is not currently pending. Then it confirms the transaction was confirmed on
 // the blockchain.
 func (e *EthTx) Perform(input models.RunInput, store *strpkg.Store) models.RunOutput {
+	if store.Config.EnableBulletproofTxManager() {
+		return e.perform(input, store)
+	} else {
+		return e.legacyPerform(input, store)
+	}
+}
+
+// TODO(sam): Move away from resuming this on every new head and do something more sensible
+func (e *EthTx) perform(input models.RunInput, store *strpkg.Store) models.RunOutput {
+	trtx, err := store.FindEthTaskRunTxByTaskRunID(input.TaskRunID().UUID())
+	if err != nil {
+		err = errors.Wrap(err, "FindEthTaskRunTxByTaskRunID failed")
+		logger.Error(err)
+		return models.NewRunOutputError(err)
+	}
+	if trtx != nil {
+		return e.checkForConfirmation(*trtx, input, store)
+	}
+	return e.insertEthTx(input, store)
+}
+
+func (e *EthTx) checkForConfirmation(trtx models.EthTaskRunTx, input models.RunInput, store *store.Store) models.RunOutput {
+	switch trtx.EthTx.State {
+	case models.EthTxConfirmed:
+		receipt := models.EthReceipt{}
+		if err := store.GetRawDB().
+			Joins("INNER JOIN eth_tx_attempts ON eth_tx_attempts.hash = eth_receipts.tx_hash AND eth_tx_attempts.eth_tx_id = ?", trtx.EthTxID).
+			First(&receipt).Error; err != nil {
+			err = errors.Wrap(err, "checkForConfirmation could not find receipt for confirmed eth_tx")
+			logger.Error(err)
+			return models.NewRunOutputError(err)
+		}
+		output, err := models.JSON{}.Add("result", receipt.TxHash.Hex())
+		if err != nil {
+			err = errors.Wrap(err, "checkForConfirmation failed")
+			logger.Error(err)
+			return models.NewRunOutputError(err)
+		}
+		return models.NewRunOutputComplete(output)
+	case models.EthTxFatalError:
+		return models.NewRunOutputError(trtx.EthTx.GetError())
+	default:
+		return models.NewRunOutputPendingOutgoingConfirmationsWithData(input.Data())
+	}
+}
+
+func (e *EthTx) insertEthTx(input models.RunInput, store *store.Store) models.RunOutput {
+	value, err := getTxData(e, input)
```

Comment:
```suggestion
	txData, err := getTxData(e, input)
```

"Value" is another field in an Etheruem transaction, the naming is a bit confusing as is.